### PR TITLE
(enh) review of logger.py; less logging in AgentController

### PR DESCRIPTION
**Pull request by tobitege**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Cleaner logger startup and reduced logging in `AgentController` when waiting.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Revised `logger.py`:
- cleaner flow of inital log level handling, also to avoid deprecated linter messages ahead of time
- actually check if a value passed in via env var is a valid mode
- `current_log_level` introduced, that can be imported elsewhere
- some methods got the log level now as parameter instead of referencing a global

In `agent_controller.py` (`AgentController`) the `_step` method is being called every second.
- removed extra debug output of `_pending_action` which lead to log spam as the same content would appear twice in the log, especially noticeable with code blocks (thought/observation) when debug mode was active

---
**Link of any specific issues this addresses**

Follow up to #3556

### CODE REVIEW

File: openhands/controller/agent_controller.py [hunk 1]
```diff
@@ -351,9 +354,22 @@ async def _step(self) -> None:
...
+            current_pending_action = self._pending_action
+
+            # New: Check if the pending action has changed
+            if (
+                not hasattr(self, '_last_pending_action')
+                or self._last_pending_action != current_pending_action
+            ):
+                logger.debug(
+                    f'[Agent Controller {self.id}] waiting for pending action...'
+                )
```
Comment (enyst) [inline openhands/controller/agent_controller.py:366]: TBH I would rather remove this completely. A developer can always add it when they need this exact info, not like we don't add plenty of extra logging when we have a particular issue to debug.

Comment (tobitege) [reply openhands/controller/agent_controller.py:366]: You mean the whole "every second" output block? It's only with debug anyways?
But personally, I wouldn't mind having it removed as well.

Comment (enyst) [reply openhands/controller/agent_controller.py:366]: Yes. I think it's just a bad place to put a logger.debug() at all. We don't need it in the vast majority of executions...

Comment (tobitege) [reply openhands/controller/agent_controller.py:366]: Agreed, I'll remove that block with the next commit.

**Comment by tobitege** [comment]
Removed the logging from AgentController's `_step()` upon `_pending_action` now. @enyst

**Review by enyst — APPROVED** [review]

**Pull request merged**